lib/prelude.dx

```diff
@@ -406,6 +409,78 @@ def finEq (n:Int) ?-> : Eq (Fin n) = MkEq \x y. ordinal x == ordinal y
 def finOrd (n:Int) ?-> : Ord (Fin n) =
   MkOrd finEq (\x y. ordinal x > ordinal y) (\x y. ordinal x < ordinal y)
 
+'## Raw pointer operations
+
+Int32Ptr : Type = %Int32Ptr
+Word8Ptr : Type = %Word8Ptr
+
+CharPtr = Word8Ptr
+
+data Ptr a:Type = MkPtr Word8Ptr
+
+-- Is there a better way to select the right instance for `storageSize`??
+data TypeVehicle a:Type = MkTypeVehicle
+
+interface Storable a:Type where
+  store : Ptr a -> a -> {State World} Unit
+  load  : Ptr a ->      {State World} a
+  storageSize : TypeVehicle a -> Int
+
+-- TODO: there's a bug preventing us inlining these definitions into the instance
+def charStore ((MkPtr ptr): Ptr Word8) (x:Word8) : {State World} Unit  = %ptrStore ptr x
```

Comment:
It's not about characters, so let's avoid pulling silly C-isms into Dex. `storeWord`/`loadWord`?